### PR TITLE
Added [:T] syntax explanation to generics tutorial.

### PR DESCRIPTION
**Pull request by n-1x**
This stumped me for a while until ElegantBeef helped me on IRC.

### CODE REVIEW

File: doc/tut2.rst [hunk 1]
```diff
@@ -534,6 +534,18 @@ iterator or type. As the example shows, generics work with overloading: the
 best match of ``add`` is used. The built-in ``add`` procedure for sequences
 is not hidden and is used in the ``preorder`` iterator.
 
+There is a special [:T] syntax when using generics with universal function call syntax:
+
+.. code-block:: nim
+  proc foo[T](i: T) =
+      discard
```
Comment (ringabout) [inline doc/tut2.rst]:
```suggestion
    discard
```

**Comment by ringabout** [comment]
Please make this example runnable: comment the wrong line and add `:test: "nim c $1`

```rst
.. code-block:: nim
    :test: "nim c $1"
```

File: doc/tut2.rst [hunk 2]
```diff
@@ -534,6 +534,18 @@ iterator or type. As the example shows, generics work with overloading: the
...
+
+  var i : int
```
Comment (ringabout) [inline doc/tut2.rst]:
```suggestion
  var i: int
```

**Comment by n-1x** [comment]
Apologies for the off formatting. I have made the change you requested.

File: doc/tut2.rst [hunk 3]
```diff
@@ -534,6 +534,19 @@ iterator or type. As the example shows, generics work with overloading: the
...
+There is a special [:T] syntax when using generics with universal function call syntax:
+
+.. code-block:: nim
+    :test: "nim c $1"
+  proc foo[T](i: T) =
+    discard
+
+  var i: int
+
+  #i.foo[int]() #Error: expression 'foo(i)' has no type (or is ambiguous)
```
Comment (ringabout) [inline doc/tut2.rst]:
```suggestion
  # i.foo[int]() # Error: expression 'foo(i)' has no type (or is ambiguous)
```

File: doc/tut2.rst [hunk 4]
```diff
@@ -534,6 +534,19 @@ iterator or type. As the example shows, generics work with overloading: the
...
+
+  i.foo[:int]() #Success
```
Comment (ringabout) [inline doc/tut2.rst]:
```suggestion
  i.foo[:int]() # Success
```

**Review by ringabout — APPROVED** [review]

File: doc/tut2.rst [hunk 5]
```diff
@@ -534,6 +534,19 @@ iterator or type. As the example shows, generics work with overloading: the
 best match of ``add`` is used. The built-in ``add`` procedure for sequences
 is not hidden and is used in the ``preorder`` iterator.
 
+There is a special [:T] syntax when using generics with universal function call syntax:
```
Comment (Araq) [inline doc/tut2.rst]: It's called the "method call syntax", there is nothing "Universal" about special casing the first argument that a function receives.

**Comment by narimiran** [comment]
I would like to also see the explanation why `:T` is needed, i.e. mention that `i.foo[int]` is parsed as `(i.foo)[int]`, while `i.foo[:int]` is parsed as `i.(foo[int])`.

File: doc/tut2.rst (hunk 5 above)
Comment (Araq) [inline doc/tut2.rst]:
```suggestion
There is a special ``[:T]`` syntax when using generics with the method call syntax:
```

**Comment by Araq** [comment]
@narimiran We can add this explanation later.

File: doc/tut2.rst (hunk 5 above)
Comment (n-1x) [reply doc/tut2.rst]: Noted. Thank you for the comment. I'm very new to Nim and wasn't exactly sure what to write there.

**Pull request merged**